src/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -792,21 +792,36 @@ public override string ToString()
         {
             if (_sddlForm == null)
             {
-                StringBuilder result = new StringBuilder();
-
                 //
                 // Typecasting of _IdentifierAuthority to a long below is important, since
                 // otherwise you would see this: "S-1-NTAuthority-32-544"
                 //
-
-                result.Append("S-1-").Append((long)_identifierAuthority);
-
-                for (int i = 0; i < SubAuthorityCount; i++)
+                // length of buffer calculation
+                // prefix = "S-1-".Length: 4;
+                // authority: long.MaxValue.ToString("D") + '-'.Length : 18+1: 20;
```

Comment:
Why the +1 here? There's no additional dash, as that's part of the subsequent items.

Reply:
And 18+1 == 20?

Reply:
And isn't long.MaxValue.ToString().Length actually 19?

Reply:
Is it possible a negative value could get in here, like long.MinValue? Then the max would be 20. Assert that it's non-negative? Or throw if it is?

Reply:
The api returns 48 bits of unsigned number which we store in a long backed enum so it can't be 64 bits and become negative. It also can't be set directly only through construction of this object which uses the api.
Yes, the 18+1 should just be 19, This is why i use computers to count for me. Max is 188.

Reply:
> The api returns 48 bits of unsigned number which we store in a long backed enum so it can't be 64 bits and become negative.

Maybe format it as `ulong` to make it easier to understand?

Reply:
ok, takes the max up to 189 even though that extra digit should never be needed it's safer to make sure if someone gets invalid input into the function it won't crash. Done.